docs/framework/configure-apps/redirect-assembly-versions.md

```diff
@@ -39,17 +44,15 @@ You can redirect compile-time binding references to .NET Framework assemblies, t
 
  If a publisher policy file exists for an assembly, the runtime checks this file after checking the assembly's manifest and app configuration file. Vendors should use publisher policy files only when the new assembly is backward compatible with the assembly being redirected.
 
- You can bypass publisher policy for your app by specifying settings in the app configuration file, as discussed in the [Bypassing publisher policy section](#bypass_PP).
-
-<a name="BKMK_Redirectingassemblyversionsattheapplevel"></a>
+ You can bypass publisher policy for your app by specifying settings in the app configuration file, as discussed in the [Bypass publisher policy](#bypass-publisher-policy) section.
 
 ## Redirect versions at the app level
```

Comment:
I wonder if we also need to cover cases where the project author does not own the application.

Tests, plugins, or libraries used by some other component.
For tests we expect folks to generate a .dll.config and most test existing unit frameworks will honor these when loading tests.
For plugins they may also honor .dll.config or they may not.  The only fool-proof mechanism for redirects is by providing bindingRedirects on AppDomain creation.

Some might try to solve this problem with AssemblyResolve handlers, but this will not work since those only are called on a failed load.  If an assembly load succeeds - either because it was loaded by another assembly, the host, or was present in the GAC an AssemblyResolve handler will not be called.